### DISCUSSION TITLE
Sujet / Thema 4 –– Community-Datenpflege / Gestion de données par la communauté

### DISCUSSION THREAD

**Discussion by rastrau** [GeoUnconference/discussions#10] [Ideen / Idées]
Resultate der Diskussion an der GeoUnconference 1 / Résultats de la discussion lors de la GeoUnconference 1

![Traffic and mobility](https://user-images.githubusercontent.com/4510815/124478904-f71ac080-dda5-11eb-8cfc-bb7a428c9d57.jpg)

![Traffic and mobility (2)](https://user-images.githubusercontent.com/4510815/124478916-f97d1a80-dda5-11eb-817c-c5ff79f758ff.jpg)

**Comment by schenardi** [comment]
Welchen Bezug kann man zur _Daten-Plattform für den Strassenverkehr in der Schweiz_ herstellen?
https://www.astra.admin.ch/astra/de/home/themen/intelligente-mobilitaet/open-data.html

Diese Veranstaltung passt hier wohl dazu: https://www.prisma-solutions.com/de/events/vdp-ch-onboarding

**Reply by rastrau** [reply]
Danke für den Hinweis, @schenardi! @datamapio, @n0rdlicht und wir haben auf Twitter mal den Bezug zu «Verkehrsnetz Schweiz» andiskutiert: https://twitter.com/GeoUnconference/status/1407963581061251074. Aber ja, die kommende «Mobilitätsdatenplattform» dürfte auch damit zusammenhängen. So wie ich @datamapio verstanden habe, trifft es Verkehrsnetz Schweiz aber noch nicht ganz (zudem dürfte es noch eine Weile gehen, bis es vorliegt).

**Comment by rogerfischer** [comment]
Quick note to myself:
Corinne Frey (am Verkehrsnetz beteiligt), Christine von derpunkt, patrick von Outdoor Active und ich waren diesmal in der Gruppe Mobilität und Verkehr. 

Aus meiner Sicht wäre es sinnvoll rasch einen Prototypen zu bauen, der die konkreten Anforderungen an einen gemeinsamen Datenpool in die Tat umsetzt.


**Einfachster Usecase: Aktuelle Gemeindegrenzen**
Da CH-Gemeinden auch während dem Jahr fusionieren (z.B. Gemeindefusion Bellinzona), aber Gemeindegrenzen nur 1 x pro Jahr zur Verfügung stehen, könnte man so ein ganz einfaches Beispiel schaffen, an dem man exemplarisch der Mehrnutzen für alle sichtbar würde. Bei jeder Abstimmung oder Wahl braucht es Gemeindegrenzen und diese sind auch historisch relevant. 

URL: gemeindegrenzen.ch 
Nutzer:innen: Alle Journalist:innen, alle Planer:innen, Behörden, d.h. eine sehr breite Nutzerbasis

Karte zeigt aktuellsten Stand der Gemeindegrenzen.
Auf einem Zeitstrahl kann man in die Vergangenheit oder in die Zukunft (geplante Fusionen)
Download: 
- LV95 und WGS84,
- GeoJSON, Shapefile


**2. Use Case, den wir vorgestellt haben** 
Aktuelles ÖV-Netz (auch mit Busstrecken)
Siehe auch:
https://github.com/datamapio/verkehrsnetz

**Reply by n0rdlicht** [reply]
gemeindegrenzen.ch wäre eine super Democase 👏 (und eine lang gesuchte Grundlage für uns).

Unser aktueller sehr kruder Proxy den wir für gemeindescan.ch verwenden basierend auf SwissBOUNDARIES3D: https://datahub.io/cividi/ch-municipalities bzw. https://github.com/cividi/ch-municipalities.

**Reply by rogerfischer** [reply]
👏 
Wir haben uns auch schon damit auseinandergesetzt, vielleicht könnte man das ja gemeinsam angehen. 
https://github.com/datamapio/geoid/blob/master/lookup.md#ch

See also:
https://github.com/datamapio/geoid/blob/master/README.md
https://github.com/datamapio/geoid/blob/master/CH/README.md

**Reply by n0rdlicht** [reply]
Absolut :) Kick-Off/Workshop "GeoUnconference-X: Shared Datapool Prototyp" im September/Oktober organisiert durch datamap und cividi?

**Reply by davidoesch** [reply]
via linked data stehen auch die Gemeindegrenzen zur Verfügung  https://ld.geo.admin.ch/data/swissBOUNDARIES3D:municipalities versioniert, zeitstrahl seit 2016 möglich über das attrribut Date  zb https://ld.geo.admin.ch/sparql/#query=PREFIX+schema%3A+%3Chttp%3A%2F%2Fschema.org%2F%3E%0APREFIX+gn%3A+%3Chttp%3A%2F%2Fwww.geonames.org%2Fontology%23%3E%0Aselect+%3FMunicipality+%3FName+%3FPopulation%0Awhere%0A%7B%0A%3FMunicipality+gn%3AfeatureCode+gn%3AA.ADM3+.%0A%3FMunicipality+schema%3Aname+%3FName+.%0A%3FMunicipality+gn%3Apopulation+%3FPopulation+.%0A%3FMunicipality+%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2Fissued%3E+%3FDate+.%0AFILTER+(%3FDate+%3D+%222019-01-01%22%5E%5Exsd%3Adate)%0A%7D%0AORDER+BY+DESC(%3FPopulation)%0ALIMIT+5&contentTypeConstruct=text%2Fturtle&contentTypeSelect=application%2Fsparql-results%2Bjson&endpoint=https%3A%2F%2Fld.geo.admin.ch%2Fquery&requestMethod=POST&tabTitle=Query&headers=%7B%7D&outputFormat=gchart&outputSettings=%7B%22chartConfig%22%3A%7B%22options%22%3A%7B%22annotations%22%3A%7B%22domain%22%3A%7B%22style%22%3A%22line%22%7D%7D%2C%22legacyScatterChartLabels%22%3Atrue%2C%22legend%22%3A%22right%22%2C%22isStacked%22%3Atrue%2C%22vAxes%22%3A%5B%7B%22title%22%3Anull%2C%22minValue%22%3Anull%2C%22maxValue%22%3Anull%2C%22viewWindow%22%3A%7B%22max%22%3Anull%2C%22min%22%3Anull%7D%2C%22useFormatFromData%22%3Atrue%2C%22logScale%22%3Afalse%7D%2C%7B%22viewWindow%22%3A%7B%22max%22%3Anull%2C%22min%22%3Anull%7D%2C%22minValue%22%3Anull%2C%22maxValue%22%3Anull%2C%22useFormatFromData%22%3Atrue%2C%22logScale%22%3Afalse%7D%5D%2C%22booleanRole%22%3A%22certainty%22%2C%22hAxis%22%3A%7B%22useFormatFromData%22%3Atrue%2C%22minValue%22%3Anull%2C%22maxValue%22%3Anull%2C%22viewWindow%22%3Anull%2C%22viewWindowMode%22%3Anull%7D%2C%22width%22%3A1887%2C%22height%22%3A600%7D%2C%22state%22%3A%7B%7D%2C%22view%22%3A%7B%22columns%22%3A%5B0%2C%7B%22sourceColumn%22%3A1%2C%22properties%22%3A%7B%22role%22%3A%22annotation%22%7D%2C%22label%22%3A%22Name%22%7D%2C2%5D%2C%22rows%22%3Anull%7D%2C%22isDefaultVisualization%22%3Afalse%2C%22chartType%22%3A%22ColumnChart%22%2C%22containerId%22%3A%22yasq_ei6autxo47za1x9u23xr_gchartWrapper%22%7D%2C%22motionChartState%22%3Anull%7D

**Reply by rogerfischer** [reply]
Kannst Du das erklären? 
Wo sehe ich z.B. die aktuellsten Gemeindegrenzen? Wo kann ich sie einfach downloaden als Geojson?

Oder wo finde ich die Gemeindegrenzen des Kanton Bern im Jahr 2016, damit so etwas darstellen kann:
https://github.com/datamapio/geoid/blob/master/CH/municipality/2016/election/enforcement_initative_bern_municipality_experimental_20160228.png

**Reply by davidoesch** [reply]
JSON-LD für eine Gemeinde
https://ld.geo.admin.ch/boundaries/municipality/geometry/351:2016?format=json

**Reply by davidoesch** [reply]
und generalisierte Gemeindegrenzen 
https://ld.geo.admin.ch/boundaries/municipality/geometry-g1/351:2016?format=json
mit https://arthur-e.github.io/Wicket/sandbox-gmaps3.html
![image](https://user-images.githubusercontent.com/4577727/132956547-ea33e705-77e2-4edd-8745-ae50157cae6e.png)

fazit:
versuch das JSON-LD zu parsen

**Reply by rastrau** [reply]
Wer pflegt diese Geometrien und wie zeitnah? Bei den klassischen Geobasisdaten zu diesem Thema gibt (bzw. zumindest: gab) es in meiner Erfahrung zum Teil Diskrepanzen zwischen dem Mutationsverzeichnis des BFE und swissBOUNDARIES<sup>3D</sup> von swisstopo.

**Reply by rogerfischer** [reply]
Immer aktuell, alles an einem Ort, benutzerfreundlich (you see what you get) - das wäre das Ziel.

**Reply by davidoesch** [reply]
Die LinkedData Geometrien in https://ld.geo.admin.ch/boundaries/municipality/ entsprechen den swissBOUNDARIES3D Geometrien und werden zeitgleich mit den übrigen Diensten nachgeführt. Oder meinst du die generalisierten?

**Reply by n0rdlicht** [reply]
@rogerfischer, @rastrau, @davidoesch Vorschlag: zwei "Geoprototyping" Sessions zu Gemeindegrenzen ggf. Verkehrsnetz als GeoUnconference "follow-up", in Zürich, ca. 2h, datamap und cividi hosted. z.b.
- GeoPrototyping 1: 13.10. oder 20.10. 17-19h, location: tbd
- GeoPrototyping 2: 24.11. oder 8.12. 17-19h, location: tbd

Meinerseits würde ich den November gerne etwas freihalten, daher am liebsten Oktober und Dezember.

**Reply by rogerfischer** [reply]
gemeindegrenzen.ch ist schon mal reserviert :)

**Reply by ilyabo** [reply]
Vielleicht wäre es sinnvoll mit Interactive Things Kontakt aufzunehmen. Sie haben diese Webseite gemacht: https://swiss-maps.vercel.app/
cc @herrstucki

**Reply by rogerfischer** [reply]
Oh, wow! I didn't know about that project from last year. 
That is very near of what I dreamed of :)

**Reply by rogerfischer** [reply]
What is however still missing :)
- The update of the Gemeindegrenzen during the years (that is the hard work that many companies do in parallel)
- and a graph that helps figure out fusions of municipalities. 
- good "REF"-Files with global ids

**Reply by jstcki** [reply]
Also on npm for convenience ;) https://www.npmjs.com/package/swiss-maps

@davidoesch wir haben ja ein Meeting am Montag zu LD 😄

**Reply by jstcki** [reply]
> * The update of the Gemeindegrenzen during the years (that is the hard work that many companies do in parallel)

@rogerfischer the swiss-maps package contains boundaries 2010 – 2021

**Reply by rogerfischer** [reply]
@herrstucki 
Meaning I cannot see the Bellinzona fusion from 2. April 2017? More specifically I cannot see the old Bellinzona in March and on April first and the new Bellinzona on the fusion day, April 2nd, and after?

See:
https://de.wikipedia.org/wiki/Bezirk_Bellinzona#Ver%C3%A4nderungen_im_Gemeindebestand
https://www.swissinfo.ch/ger/direktedemokratie/schauplatz-schweiz-8-_fusion-von-13-gemeinden-zu-neuem-bellinzona/42927484

**Reply by jstcki** [reply]
@rogerfischer ah yes, sorry I misread your comment … "during" is the key word 😅 . Not sure if those versions are in the BFS/FSO shapefiles that we're using – they are for 2021 but not sure for other years. Also not sure if the LD from swisstopo contains these, they seem to be updated yearly only?

**Reply by rogerfischer** [reply]
@herrstucki This is the biggest pain point in everyone having to deal with elections and wanting to display the results on a map. Maps are only available at the start of the year (or end of the year) and there are no updates in between. Elections and municipality fusions also happen during the year.

And anyone trying to compare two elections separated by a few years or doing a time series will immediately understand how the current state of affairs is unsatisfying.

Also tracking certain municipalities over the years becomes very hard. 
They might change there name more than once. So we need some kind of municipal genealogy too.

**Reply by rastrau** [reply]
Agreed. I think a feasible route is to use swissBOUNDARIES3D of year x - 1 and apply the information from https://www.bfs.admin.ch/bfs/de/home/grundlagen/agvch.html onto the dataset oneself. In https://www.agvchapp.bfs.admin.ch/de/projects, one can see forthcoming municipal mutations (as best as they are known, I imagine). I'm not sure how frequently the BFS data is updated (current data stems from Aug 2021). 

There is also a REST servcice at https://www.bfs.admin.ch/bfs/de/home/grundlagen/agvch/historisiertes-gemeindeverzeichnis.assetdetail.15224054.html. Here, oddly the REST service is OPEN-BY-ASK which doesn't make sense to me. From the documentation it is unclear, if the REST service also covers the forthcoming mutations mentioned above.

In the past, we have used the BFS mutation listings in order to transform statistical data at an "old" municipal level to the current set of municipalities. In that process, I think we have occasionally/rarely also seen that the BFS information did not match the swisstopo data (e.g., executed municipal fusion that is not yet shown in swisstopo data from a later year).

**Reply by davidoesch** [reply]
via STAC swissboundaries monthly versions are available as well 2021 onward https://data.geo.admin.ch/browser/index.html#/2fFdsivmHi9SSzqeVRNdGebNsRxQ5iXV58mBhj2eK8zC1PzCMkrFW3TAX5?t=collections

**Reply by RafBov** [reply]
A lot of information exists here : https://www.bfs.admin.ch/bfs/de/home/grundlagen/agvch/historisiertes-gemeindeverzeichnis.html 

There are different dataset already available. The topic is quiet complicated.. alonside with fusion there are other changes (new / better survey, new definition..) . Gemeindegrenzen seems an easy topic at the first sight..

**Comment by rastrau** [comment]
Die Gruppe hat ihre Arbeiten auf [einem Miro-Board](https://miro.com/app/board/o9J_lxjftic=/) dokumentiert. Hier ein Screenshot, damit alle Resultate im Thread enthalten sind:

![GeoUnconference](https://user-images.githubusercontent.com/4510815/133229577-a9b95dc0-3fe0-48e1-a8c6-414e53098d0f.jpg)

**Reply by rogerfischer** [reply]
Danke Ralph!

**Reply by rastrau** [reply]
Danke Euch vielmals für Eure Überlegungen!

**Reply by rogerfischer** [reply]
@n0rdlicht @herrstucki @ilyabo @rastrau @davidoesch 
GeoPrototyping 1: 20.10. 17-19h, location: tbd?

**Reply by rastrau** [reply]
Falls gewünscht könnten wir gerne mal eine Session bei EBP am Stadelhofen hosten. Am 20. und 21.10. haben wir aber leider mit einem Open House-Anlass für Studierende bereits "Full House"... Ein anderes Mal könnten wir Facilities bieten.

**Reply by ilyabo** [reply]
Wäre Freitag 29.10 keine gute Option? Ich könnte Abends nur an einem Freitag mitmachen und von 10.10 bis 22. sind Schulferien, wir sind dann wahrscheinlich weg.

**Reply by rogerfischer** [reply]
Freitag 29.10 am Nachmittag, 16-18 Uhr / Open End, würde mir auch gehen.
Euch? @rastrau @n0rdlicht @davidoesch @herrstucki ?

**Reply by rogerfischer** [reply]
Hättet ihr da Zeit?
@rastrau @n0rdlicht @davidoesch @herrstucki @ilyabo @schenardi @RafBov 
Freitag 29.10 am Nachmittag, 16-18 Uhr / Open End

**Reply by n0rdlicht** [reply]
Sowohl 20. als auch 29. nachmittags passen bei mir...

**Reply by rastrau** [reply]
29. ginge 👍 (könnten wohl auch hosten am Stadelhofen; wenn gewünscht, prüfe ich die Verfügbarkeit von Räumen)

**Reply by rogerfischer** [reply]
Blöd, jetzt geht mir der 29.10 doch nicht wegen DinaCon. 
Sollen wir ein Doodle machen?
Oder geht Euch spontan grad der 5. November?
@rastrau @n0rdlicht @davidoesch @herrstucki @ilyabo @schenardi @RafBov

**Reply by n0rdlicht** [reply]
5.11. würde bei mir passen. Gerne sonst per Doodle

**Reply by rogerfischer** [reply]
https://doodle.com/poll/hr4s2cq635zshuqh

**Reply by rogerfischer** [reply]
Sieht nach 19.11 aus. @rastrau - wäre ein hosting bei euch möglich?
 
@n0rdlicht @davidoesch @herrstucki @ilyabo @schenardi @RafBov

**Reply by rastrau** [reply]
Absolut! Habe mal zwei Sitzungszimmer an der [Mühlebachstrasse 11 in Zürich](https://s.geo.admin.ch/939b496e08) reserviert. 
Bitte gebt noch kurz Bescheid, wer effektiv teilnimmt, damit ich das Sitzungszimmer entweder in gross oder in sehr gross reservieren kann:

- [x] Roger Fischer / @rogerfischer 
- [x] Ilya Boyandin / @ilyabo 
- [x] Thorben Westerhuys / @n0rdlicht
- [ ] David Oesch / @davidoesch 
- [x] Ralph Straumann
- [ ] Raphaël Bovier? / @RafBov
- [ ] Markus Schenardi? / @schenardi 
- [ ] Jeremy Stucki? / @herrstucki
- [ ] weitere...?

**Reply by davidoesch** [reply]
Dear all-- auch der 19.11 geht mir nicht ----

**Reply by schenardi** [reply]
sorry, wegen Ferien sehr offline.... 19.11. kann ich leider auch nicht.

**Comment by schenardi** [comment]
Ich "höre" hier schon eine Weile zu und will euch gerne einen Input resp. meine Meinung dazu geben zu diesem auf den ersten Blick einfachen Thema "Gemeindegrenzen". Aber auf den zweiten Blick ist das hochkomplex.

* Bevor hier unendlich viel Zeit investiert wird in die Aufbereitung von historischen Daten würde ich mir die Anwendungsfälle nochmals klar notieren. Was soll der User am Ende mit den Daten tun können? Was ev. die Maschine? Daraus ableiten, wieviele alte Daten wirklich noch nötig sind, in welcher Form
Vorlage ist bekannt: #13 
* Aus meiner Sicht müssten Swisstopo oder die Kantone die Daten historisiert aber wohl auch perspektivisch (bevorstehende Fusionen) bereitstellen. Inwieweit das schon erfolgt ist mir nicht ganz klar. @davidoesch hat diverse Beispiele eingebracht, unter anderem ein Hinweis, dass zumindest ab 2021 Monatsstände verfügbar sind? Genügt es, wenn ab "jetzt" das besser gemacht wird oder muss nachgebessert werden?
* Daten müssten über die üblichen Schnittstellen bereitgestellt werden. WMS mit Historie, WFS, aber vor allem STAC-API scheinen mir geeignet. Für ArcGIS User eine GDB mit Archiving aktiviert wäre durchaus auch interessant oder gar dasselbe als public ArcGIS Online Service (Jahresstände existieren bereits in ArcGIS online von Esri CH - aber nur limitierte Nutzung erlaubt, warum müsste nachgefragt werden)

**Reply by rogerfischer** [reply]
Gemeindegrenzen sind meiner Meinung nach zentral für fast alle Geodienste, ganz viele weitere Datensätze basieren auf Ihnen oder beziehen sich auf sie (z.B. Verkehrsknoten). 

Uns (Datamap) persönlich interessieren eigentlich nur webbasierte Anwendungen, hauptsächlich im WGS84 Format (TopoJSON, GeoJSON). Wir brauchen dafür noch nicht einmal ein API, einfach die Grenzen in einem Vektorformat zum Download.


Usecase 1:
Für jede Abstimmung braucht es Gemeindegrenzen, die up-to-date sind. Und das kommt in der Schweiz ja bekanntlich öfters vor. D.h. alle Journalist:innen in der Schweiz, die online Karten publizieren, wären da sicher super dankbar. 

Usecase 2: 
=> Spielt in Public Transport Geo Editor hinein. Siehe: https://github.com/datamapio/verkehrsnetz#woran-wir-arbeiten
Wenn Gemeindegrenzen wechseln, schlägt das z.B. auch auf Verkehrsknoten durch.

**Reply by RafBov** [reply]
Here a liste of available Dataset (I work for swisstopo) 

From BFS Generalized boundaries  : https://www.bfs.admin.ch/bfs/de/home/dienstleistungen/geostat/geodaten-bundesstatistik/administrative-grenzen/generalisierte-gemeindegrenzen.html 
Available from 2001 - now. I dataset per year with the state 1. January of the year.  No changes are made in the middle of the year. 

from swisstopo : swissBOUNDARIES3D : https://www.swisstopo.admin.ch/de/geodata/landscape/boundaries3d.html or in the stac api
For the moment just the data of 2021 are available. Since 2020 swisstopo publish a new dataset wenn a fusion takes place (the day of the official fusion).  Older state could be provided from 2003 onwards. No generalisation of the geometries, but if you go back changes happens in the model and in the way the boundaries are represented (f.eg. in some cantons the lakes are divided in municipalities in other not.. ) 

If you want to monitor a municipality over the time, the best thing is to use the historisierte Gemeindenummer. unfortunately this number is not as attribute in the geodata yet. The numbers are availabe here https://www.bfs.admin.ch/bfs/de/home/grundlagen/agvch/historisiertes-gemeindeverzeichnis.html 

If you need more info, just let me know.

**Comment by ktk** [comment]
The data is available for some time, you simply have to run a SPARQL query to the Swisstopo SPARQL endpoint. I prepared one below with different scenarios you can enable/disable. If anything is unclear, please let me know, happy to help.

Click on "Table" if you want to see the result in a table. The shapes are in [Well Known Text](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) format, I guess you know what to do with that.

Feel free to play with the query and enable/disable different parts in there to tailor it to your needs.

![Screen Shot 2021-10-08 at 10 06 59](https://user-images.githubusercontent.com/583021/136521217-217923e6-f031-440b-b810-2f230461d831.png)

I also plan a blog post about that for some more details, will post here once I published it.

[SPARQL](https://linked-data-training.zazuko.com/SPARQL/#1) is a very generic API for RDF/Linked Data. It's easy to do complex things which you couldn't do in a plain HTTP API (and GraphQL for that matter).

The query is [here](https://ld.geo.admin.ch/sparql/#query=PREFIX+xsd%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%3E%0APREFIX+dcterm%3A+%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2F%3E%0APREFIX+rdf%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23%3E%0APREFIX+rdfs%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%3E%0APREFIX+dcterms%3A+%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2F%3E%0APREFIX+geo%3A+%3Chttp%3A%2F%2Fwww.opengis.net%2Font%2Fgeosparql%23%3E%0APREFIX+skos%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2004%2F02%2Fskos%2Fcore%23%3E%0APREFIX+schema%3A+%3Chttp%3A%2F%2Fschema.org%2F%3E%0APREFIX+gn%3A+%3Chttp%3A%2F%2Fwww.geonames.org%2Fontology%23%3E%0A++%0ASELECT+*+WHERE+%7B%0A%7B%0A++%23+This+is+fetching+the+latest+year+of+the+data.+This+can+be+hardcoded+as+well%2C+see+other+query.%0A++SELECT+(max(%3Fissued)+AS+%3FmostRecentYear)+WHERE+%7B%0A++++%3Chttps%3A%2F%2Fld.geo.admin.ch%2Fboundaries%2Fcanton%2F1%3E+dcterms%3AhasVersion%2Fdcterms%3Aissued+%3Fissued.%0A++%7D%0A%7D%0A%0A%3Fcanton+gn%3AfeatureCode+gn%3AA.ADM3+%3B%0A++dcterms%3AhasVersion+%3FgeomuniVersion+.%0A%0A%23+If+you+want+an+older%2Fspecific+year%2C+you+can+comment+lines+11-16+and+enable+this+one+below.%0A%23+See+https%3A%2F%2Fld.geo.admin.ch%2Fboundaries%2Fmunicipality%2F1001+to+get+an+idea+of+what+years+are+available%0A%23++BIND(+xsd%3Adate('2021-01-01')+AS+%3FmostRecentYear+)%0A++%0A%3FgeomuniVersion+dcterms%3Aissued+%3FmostRecentYear+%3B%0A++++++++++++++++%0A++schema%3Aname+%3FmunicipalityName+%3B+%23+This+is+the+official+name+of+that+year+for+the+municipality%0A++%3Chttps%3A%2F%2Fld.geo.admin.ch%2Fdef%2FbfsNumber%3E+%3FmunicipalityId+%3B+%23+the+non-versioned+ID+of+the+municipality+(Gemeindenummer)%0A++++%0A++%23+This+is+a+simplified+version+of+the+shape+as+Well+Known+Text%2C+use+this+if+you+don't+need+all+details%0A++geo%3AhasGeometry%2Fgeo%3AasWKT+%3Fwkt+.%0A++%0A++%23+This+is+the+full+detail+version+of+the+shapes.+This+is+a+*lot*+bigger%2C+only+use+this+if+you+need+the+details%0A%23++geo%3AdefaultGeometry%2Fgeo%3AasWKT+%3Fwkt+.+%0A++%0A++%23+If+you+enable+this+line%2C+you+will+get+a+label+on+each+municipality+on+the+map.+But+that's+a+bit+heavy+to+render%0A%23++BIND(+CONCAT(+%3FmunicipalityName%2C+'+('%2C+%3FmunicipalityId%2C+')')+AS+%3FwktLabel+)%0A%7D%0A&contentTypeConstruct=text%2Fturtle&contentTypeSelect=application%2Fsparql-results%2Bjson&endpoint=https%3A%2F%2Fld.geo.admin.ch%2Fquery&requestMethod=POST&tabTitle=Alle+Kantone&headers=%7B%7D&outputFormat=leaflet)

You can copy/paste the text into a file (`municipalities.rq`) and get the content back via curl:

    curl -H "Accept: text/csv" --data-urlencode query@municipalities.rq https://ld.geo.admin.ch/query

SPARQL offers result sets in multiple forms: [CSV/TSV](http://www.w3.org/TR/sparql11-results-csv-tsv/), [XML](http://www.w3.org/TR/rdf-sparql-XMLres/) and [JSON](http://www.w3.org/TR/sparql11-results-json/), simply provide one of these `Accept` headers if you don't want CSV:

* JSON: `application/sparql-results+json`
* XML: `application/sparql-results+xml`

This should be easy to use in any programming language that way. In JavaScript, we recommend to use [our SPARQL library](https://github.com/zazuko/sparql-http-client) 

PS: That query will likely change in the future as some namespaces are refactored right now. Watch this space for news.

**Comment by DiegoVonMentlen** [comment]
Resultate der Diskussion an der GeoUnconference 4 / Résultats de la discussion lors de la GeoUnconference 4

![image](https://user-images.githubusercontent.com/4510815/196648175-a65b11a2-953d-4492-af62-760fc16b810d.png)

![unconference1](https://user-images.githubusercontent.com/115772183/195768614-eb5fee54-b1d9-45d8-b707-cb985d0f1b55.jpg)

![unconference2](https://user-images.githubusercontent.com/115772183/195768631-ce0b451b-8b5c-47a3-9c3f-559c34ec6382.jpg)

![unconference3](https://user-images.githubusercontent.com/115772183/195768636-120bcf1a-a780-46b6-b8b3-781cdbbef2ac.jpg)

![unconference4](https://user-images.githubusercontent.com/115772183/195768641-e5bd5411-97bc-4e45-9114-92769d161420.jpg)

![unconference5](https://user-images.githubusercontent.com/115772183/195768654-3ec24017-ccfc-4993-a330-a4628c7ca589.jpg)

![unconference6](https://user-images.githubusercontent.com/115772183/195768661-f72691b3-91f2-4a50-bad8-6b8978ee95f9.jpg)
